xwiki-pro-macros-ui/src/main/resources/XWiki/Macros/Taglist.xml

```diff
@@ -0,0 +1,641 @@
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </id>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Macro name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <priority>
+        <disabled>0</disabled>
+        <name>priority</name>
+        <number>11</number>
+        <numberType>integer</numberType>
+        <prettyName>Priority</prettyName>
+        <size>10</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.NumberClass</classType>
+      </priority>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#macro(excludeTags $tagSet $excludedTagsList)
+  #foreach ($excludedTag in $excludedTagsList)
+    #set ($discard = $tagSet.remove($excludedTag))
+  #end
+#end
+
+#macro(createLists $bins, $baseUrl)
+  &lt;ul class="glossaryListRoot"&gt;
+  #foreach($key in $bins.keySet())
+    &lt;li&gt;
+      &lt;div class="glossaryBinTitle"&gt;$key&lt;/div&gt;
+      &lt;ul class="glossaryBinRoot"&gt;
+        #foreach($element in $bins.get($key))
+          &lt;li class="glossaryBinElement"&gt;&lt;a href="$baseUrl?do=viewTag&amp;tag=$element"&gt;$element&lt;/a&gt;&lt;/li&gt;
+        #end
+      &lt;/ul&gt;
+    &lt;/li&gt;
+  #end
+  &lt;/ul&gt;
+#end
+
+#macro(createHtmlStructure, $tags, $limit)
+  #set ($baseUrl = $xwiki.getDocument('Main.Tags').getURL())
+  #set ($bins = $services.taglist.getBins1($tags))
```

Comment:
```suggestion
  #set ($bins = $services.taglist.getBins($tags))
```
You should run a test on a fresh instance to make sure the macro works as intended.